File: 012)CatchTheOnePiece/app/src/main/java/com/example/catchtheonepiece/CatchAdapter.java
package com.example.catchtheonepiece;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.example.catchtheonepiece.databinding.CardDesignBinding;

import java.util.List;

public class CatchAdapter extends RecyclerView.Adapter<CatchAdapter.CardViewDesignObjectsHolder>{
    private Context mContext;
    private List<Character> onePiecesList;

    public CatchAdapter(Context mContext, List<Character> onePiecesList) {
        this.mContext = mContext;
        this.onePiecesList = onePiecesList;
    }

    @NonNull
    @Override
    public CardViewDesignObjectsHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(mContext);
        CardDesignBinding design = DataBindingUtil.inflate(layoutInflater, R.layout.card_design, parent, false);
        return new CardViewDesignObjectsHolder(design);
    }

    @Override
    public void onBindViewHolder(@NonNull CardViewDesignObjectsHolder holder, int position) {
        Character character = onePiecesList.get(position);
        CardDesignBinding design = holder.design;
        design.textViewCatch.setText(character.getCharacterName());
        design.cardViewLine.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(mContext, CatchActivity.class);
                intent.putExtra("character", character);
                mContext.startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        if (onePiecesList == null){
            return 0;
        }else {
            return onePiecesList.size();
        }
    }

    public class CardViewDesignObjectsHolder extends RecyclerView.ViewHolder{
        private CardDesignBinding design;

        public CardViewDesignObjectsHolder(CardDesignBinding design) {
            super(design.getRoot());
            this.design = design;
        }
    }
}

File: 053)CatchTheOnePieceKotlin/app/src/main/java/com/elyesasimsek/catchtheonepiecekotlin/CatchAdapter.kt
package com.elyesasimsek.catchtheonepiecekotlin

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.elyesasimsek.catchtheonepiecekotlin.databinding.CardDesignBinding

class CatchAdapter :RecyclerView.Adapter<CatchAdapter.CardViewDesignObjectsHolder>{
    private lateinit var context: Context
    private lateinit var onePieceList: List<Character>

    constructor(context: Context, onePieceList: List<Character>) {
        this.context = context
        this.onePieceList = onePieceList
    }

    class CardViewDesignObjectsHolder(val binding: CardDesignBinding): RecyclerView.ViewHolder(binding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CardViewDesignObjectsHolder {
        val layoutInflater = LayoutInflater.from(context)
        val binding = DataBindingUtil.inflate<CardDesignBinding>(layoutInflater, R.layout.card_design, parent, false)
        return CardViewDesignObjectsHolder(binding)
    }

    override fun getItemCount(): Int {
        if (onePieceList == null){
            return 0;
        }else{
            return onePieceList.size
        }
    }

    override fun onBindViewHolder(holder: CardViewDesignObjectsHolder, position: Int) {
        val character = onePieceList[position]
        val binding = holder.binding
        binding.textViewCatch.text = character.characterName
        binding.cardViewLine.setOnClickListener {
            val intent = Intent(context, CatchActivity::class.java)
            intent.putExtra("character", character)
            context.startActivity(intent)
        }
    }
}